packages/ember-data/lib/system/model/model.js

```diff
@@ -295,7 +295,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @property id
     @type {String}
   */
-  id: null,
+  id: Ember.computed.alias('_internalModel.id'),
```

Comment:
Would that work for setting the id as well? The main cause of the regression is that the id is not being set correctly when it is modified after creation.